src/save_index.hpp

```diff
@@ -134,5 +136,6 @@ class save_index_class : public std::enable_shared_from_this<save_index_class>
 	 * write_save_index() and delete() are no-ops.
 	 */
 	bool read_only_;
+	bool clean_up_index_;
```

Comment:
A comment would be good - while the commit message is good, assume people will later be looking at this code without referring to the Git history.

Maybe `/** Flag to only run the clean_up_index method once */`